### ISSUE TITLE
Make public, document and standardize the API of the `queryInterface.showAllTables` method

### ISSUE THREAD

**Issue by papb** [sequelize/sequelize#11451]
Currently the `queryInterface.showAllTables` method is marked as `@private` and therefore does not appear in the auto-generated docs. The jsdoc placed on it deserves improvement as well.

https://github.com/sequelize/sequelize/blob/d5e4a10f4e8dc1e37fe46678f04bbfa66b1299cc/lib/query-interface.js#L407-L421

Its API is also inconsistent across dialects:
* Some dialects return an array of strings (table names)
* Some dialects return an array of objects of the form `{ tableName, schema }`

I think this should be changed to always return `{ tableName, schema? }`. Of course this would be a breaking change though.

**Comment by joaoe** [comment]
All RDBMSs support schemas. Sqlite does not. So for sqlite perhaps: a) keep old behavior to break less existing code using sqlite databases currently b) return `{schema:null}` to normalize the API. I don't have any strong opinions.

**Comment by joaoe** [comment]
Feature request: extend `showAllTables()` to accept a `schemaName` as argument and list only tables for that schema.

**Comment by papb** [comment]
@joaoe Thanks!! I prefer something like option `b`, but instead of returning `{tableName:...,schema:null}` I think simply returning `{tableName: 'abc'}` is better (no schema at all)

**Comment by misos1** [comment]
Maybe something similar should be also done for showAllSchemas. It is public in documentation where it returns just Promise\<Array\> but content differ per dialect.

mysql (lists only actual database with tables):
``` js
[
  { Tables_in_database1: 'table_1' },
  { Tables_in_database1: 'table_2' }
]
```

mariadb (lists all databases):
``` js
[ 'database1', 'database2' ]
```

**Comment by papb** [comment]
@misos1 Indeed! Thanks. Can you open a new issue to track this? And link to this one? Thanks

**Comment by nodejh** [comment]
@papb Hi, what properties does the `schema` have? 

Sometimes I want to get some information (e.g.:  `charset` `comment`) passed to `queryInterface.createTable` when I use `queryInterface.showAllTables` method. 

```js
  /**
   * Create a table with given set of attributes
   *
   * ```js
   * queryInterface.createTable(
   *   'nameOfTheNewTable',
   *   {
   *     // ...
   *   },
   *   {
   *     engine: 'MYISAM',    // default: 'InnoDB'
   *     charset: 'latin1',   // default: null
   *     schema: 'public',    // default: public, PostgreSQL only.
   *     comment: 'my table', // comment for table
   *     collate: 'latin1_danish_ci' // collation, MYSQL only
   *   }
   * )
   * ```
   */
  createTable(tableName, attributes, options, model) {
    // ...
 }
```

Feature request:  I want `showAllTables` or a new method `showTable(tableName)` returning something about the table.

**Comment by papb** [comment]
@nodejh I don't know :sweat_smile: if you could investigate that that would be great :)
